CMakeLists.txt

```diff
@@ -138,7 +150,8 @@ target_compile_definitions(linux_specific INTERFACE _FILE_OFFSET_BITS=64 PAGE_GU
     $<$<BOOL:${X11_FOUND}>:VK_USE_PLATFORM_XLIB_KHR>
     $<$<BOOL:${X11_Xrandr_FOUND}>:VK_USE_PLATFORM_XLIB_XRANDR_EXT>
     $<$<BOOL:${XCB_FOUND}>:VK_USE_PLATFORM_XCB_KHR>
-    $<$<BOOL:${WAYLAND_FOUND}>:VK_USE_PLATFORM_WAYLAND_KHR>)
+    $<$<BOOL:${WAYLAND_FOUND}>:VK_USE_PLATFORM_WAYLAND_KHR>
+    $<$<BOOL:${HEADLESS}>:VK_USE_PLATFORM_HEADLESS>)
```

Comment:
Does this need to be restricted to Linux, or can it also be enabled for Windows?  Currently, HEADLESS is set to true for both Windows and Linux, so the headless files are included in the Windows build, but the `--wsi headless` option is unavailable because VK_USE_PLATFORM_HEADLESS isn't defined for Windows.

Reply:
> Does this need to be restricted to Linux
No, updated now, hopefully the new solution is ok (let me know if you want me to change it).